test/terraform_aws_nextjs_serverless_test.go

```diff
@@ -0,0 +1,166 @@
+	if aliasURL == "" {
+		aliasURL = resultStat.Get("static-deploy.next_distribution.domain_name").String()
+	}
+
+	// Stage 1
+	test_structure.RunTestStage(t, "serverHealthCheck", func() {
+		hitTheFrontPage(t, aliasURL)
+	})
+
+	// Stage 2
+	test_structure.RunTestStage(t, "publicAssetsCheck", func() {
+		getIcons(t, aliasURL)
+	})
+
+	// Stage 3
+	// I could've checked the S3 buckets to see if these resources are uploaded there, too
+	// but assuming that terraform apply finished with success, I just need to check the build
+	// output of the NextJS app: required-server-files.json and build-manifest.json
+	// fileList := getRequiredServerFiles(aliasURL) //apparently these files do not exist in the static asset bucket
+	// fileList = append(fileList, getStaticAssets(aliasURL)...)
+	fileList := getStaticAssets(aliasURL)
+	fmt.Println(fileList)
+
+	test_structure.RunTestStage(t, "staticAssetsCheck", func() {
+		checkIfFilesExist(t, fileList, aliasURL)
+	})
+}
+
+func deployTerraform(t *testing.T, workingDir string, varfiles []string) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Reconfigure:  true,
+		VarFiles:     varfiles,
+	})
+	terraform.InitAndApply(t, terraformOptions)
+}
+
+func undeployTerraform(t *testing.T, workingDir string, varfiles []string) {
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Reconfigure:  true,
+		VarFiles:     varfiles,
+	})
+	terraform.Destroy(t, terraformOptions)
+}
+
+// the front page should contain next.js in the raw text and it should respond with 200
+func hitTheFrontPage(t *testing.T, aliasURL string) {
+	tlsConfig := tls.Config{}
+	code, body := http_helper.HttpGet(t, "https://"+aliasURL, &tlsConfig)
+	assert.Equal(t, code, 200)
+	assert.Contains(t, strings.ToLower(body), "next.js")
+}
+
+// hitting icons, which are public assets
+func getIcons(t *testing.T, aliasURL string) {
+	tlsConfig := tls.Config{}
+	code, _ := http_helper.HttpGet(t, "https://"+aliasURL+"/assets/vercel.svg", &tlsConfig)
+	assert.Equal(t, code, 200)
+	code, _ = http_helper.HttpGet(t, "https://"+aliasURL+"/assets/next.svg", &tlsConfig)
+	assert.Equal(t, code, 200)
+	code, _ = http_helper.HttpGet(t, "https://"+aliasURL+"/favicon.ico", &tlsConfig)
+	assert.Equal(t, code, 200)
+}
+
+// Accepts an array of strings as fileList, hits the aliasURL with these paths, expects 200
+func checkIfFilesExist(t *testing.T, fileList []string, aliasURL string) {
+	tlsConfig := tls.Config{}
+	exists := true
+	for _, files := range fileList {
+		fmt.Println("https://" + aliasURL + files)
+		code, _ := http_helper.HttpGet(t, "https://"+aliasURL+files, &tlsConfig)
+		if code == 200 {
+			exists = true
+		} else {
+			exists = false
+			break
+		}
+	}
+	assert.True(t, exists)
+}
+
+// test stage for the files mentioned in required-server-files.json
```

Comment:
It's not a test stage but a helper function. It's also quite clear from the function name that it's getting the required files. pls remove unused param.

Reply:
Fixed.